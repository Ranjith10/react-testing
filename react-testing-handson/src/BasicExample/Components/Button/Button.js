import React from 'react'

import './Button.css'

const Button = ({label}) => {
    return (
        <>
        <div>React Testings</div>        
        <button data-testid='button'>
            {label}            
        </button>
        </>
    )
}

export default Button
