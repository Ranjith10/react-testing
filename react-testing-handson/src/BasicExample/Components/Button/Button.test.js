import React from 'react'
import ReactDom from 'react-dom'
import {cleanup, render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import Button from './Button'

afterEach(cleanup)

test('button renders without crashing', () => {
    let element = document.createElement('div')
    ReactDom.render(<Button/>, element)
})

test('button renders correctly', () => {
    let {getByTestId} = render(<Button label='Click Me' />)
    expect(getByTestId('button')).toHaveTextContent('Click Me')
})

test('button renders correctly 1', () => {
    let {getByTestId} = render(<Button label='Save' />)
    expect(getByTestId('button')).toHaveTextContent('Save')
})

test('snapshot testing sample', () => {
    const tree = renderer.create(<Button label='Testing React'/>).toJSON()
    expect(tree).toMatchSnapshot()
    // expect(container.firstChild).toMatchInlineSnapshot(`<div>React Testing</div>`)
})