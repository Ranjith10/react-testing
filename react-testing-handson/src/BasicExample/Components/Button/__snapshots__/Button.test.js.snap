// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing sample 1`] = `
Array [
  <div>
    React Testings
  </div>,
  <button
    data-testid="button"
  >
    Testing React
  </button>,
]
`;
